### PR TITLE
Add density tests

### PR DESCRIPTION
**Pull request by edknock**
Adding tests for the C++ density functions, truncated normal is outstanding because that possibly needs a bit of a rethink.

Also potentially there needs to be some thought about corner cases on some of the continuous distributions where it could be considered that the distribution becomes a point mass at 0 of infinite density

### CODE REVIEW

File: inst/include/monty/random/density.hpp [hunk 1]
```diff
@@ -145,6 +145,8 @@ __host__ __device__ T beta_binomial_ab(T x, T size, T a, T b, bool log) {
   T ret;
   if (x == 0 && size == 0) {
     ret = 0;
+  } else if (x > size) {
+    ret = -random::utils::infinity<T>();
```
Comment (edknock) [inline inst/include/monty/random/density.hpp:149]: Ensure -Inf outside domain

File: inst/include/monty/random/density.hpp [hunk 2]
```diff
@@ -288,16 +319,27 @@ __host__ __device__ T weibull(T x, T shape, T scale, bool log) {
...
                 "weibull should only be used with real types");
 #endif
-  const auto ret = monty::math::log(shape) + (shape - 1) * monty::math::log(x) -
-    shape * monty::math::log(scale) - (x / scale)^shape;
+  T ret;
+  if (x < 0) {
+    ret = -random::utils::infinity<T>();
+  } else {
+    ret = monty::math::log(shape) + (shape - 1) * monty::math::log(x) -
+      shape * monty::math::log(scale) - monty::math::pow(x / scale, shape);
```
Comment (edknock) [inline inst/include/monty/random/density.hpp:327]: Note here we also correct the power setup at the end
